### DISCUSSION TITLE
#include <core_version.h> in Settings.h

### DISCUSSION THREAD

**Discussion by chrwh** [HomeSpan/HomeSpan#744] [Q&A]
Hi,
In order to make HomeSpan compile with M5stack, I had to remove #include <core_version.h> in Settings.h, since M5stack board manager dosent include this file. the ESP version is 2.1.0 and all seems to be working as intended.
Is the core_version.h include only used to check that ESP is above V2.x.x
Best regards,
Christian

**Comment by HomeSpan** [comment]
I'm unaware of any M5stack-specific board manager.  The latest version of the Arduino-ESP32 library is 2.0.14, and it already includes configurations for variety of M5Stack boards.  Sounds like you installed the Arduino-ESP32 board manager incorrectly.  

See https://github.com/HomeSpan/HomeSpan/blob/master/docs/Solutions.md#compiler-error---core_versionh-file-not-found for details.

**Reply by chrwh** [reply]
M5Stack has there own board manager: [https://docs.m5stack.com/en/quick_start/arduino](url) so I prefere to use that, since it's M5Stacks own, and it has more settings. But for some reason they have not included core_version.h. 
It is working without modifications when I use the ESP32 Arduino board manager 2.0.14, because it has the core_version.h

I just wanted to know if all core_version.h is used for is for checking that we are above version 2.0.0

Best regards,
Christian

![Arduino 1 8 16](https://github.com/HomeSpan/HomeSpan/assets/6816822/e8a46c24-706b-402c-951b-13e5c7af559f)

**Reply by HomeSpan** [reply]
Yes, that file is only used to check the version.
